packages/event-migration/src/read-models-migration-check.ts

```diff
@@ -0,0 +1,157 @@
+    SCALA_READMODEL_DB_PORT: z.coerce.number().min(1001),
+    NODE_READMODEL_DB_HOST: z.string(),
+    NODE_READMODEL_DB_NAME: z.string(),
+    NODE_READMODEL_DB_USERNAME: z.string(),
+    NODE_READMODEL_DB_PASSWORD: z.string(),
+    NODE_READMODEL_DB_PORT: z.coerce.number().min(1001),
+  })
+  .transform((c) => ({
+    readModelCollection: c.READ_MODEL_COLLECTION,
+    scalaReadModelConfig: {
+      readModelDbHost: c.SCALA_READMODEL_DB_HOST,
+      readModelDbName: c.SCALA_READMODEL_DB_NAME,
+      readModelDbUsername: c.SCALA_READMODEL_DB_USERNAME,
+      readModelDbPassword: c.SCALA_READMODEL_DB_PASSWORD,
+      readModelDbPort: c.SCALA_READMODEL_DB_PORT,
+    },
+    nodeReadModelConfig: {
+      readModelDbHost: c.NODE_READMODEL_DB_HOST,
+      readModelDbName: c.NODE_READMODEL_DB_NAME,
+      readModelDbUsername: c.NODE_READMODEL_DB_USERNAME,
+      readModelDbPassword: c.NODE_READMODEL_DB_PASSWORD,
+      readModelDbPort: c.NODE_READMODEL_DB_PORT,
+    },
+  }));
+
+type Config = z.infer<typeof Config>;
+
+const config: Config = {
+  ...Config.parse(process.env),
+};
+
+const Identifiable = z
+  .object({
+    id: z.string(),
+  })
+  .passthrough();
+type Identifiable = z.infer<typeof Identifiable>;
+
+async function main(): Promise<void> {
+  const scalaReadModelDb = connectToReadModelDb(config.scalaReadModelConfig);
+  const nodeReadModelDb = connectToReadModelDb(config.nodeReadModelConfig);
+
+  const differences = await getReadModelsCollectionDataDifferences(
+    scalaReadModelDb,
+    nodeReadModelDb,
+    config.readModelCollection
+  );
+
+  if (differences.length > 0) {
+    console.warn(`Differences found, red is scala, green is node:`);
+    differences.forEach(([scala, node]) => {
+      consoleStringDiffs(
+        JSON.stringify(scala, null, 2),
+        JSON.stringify(node, null, 2)
+      );
+    });
+    process.exit(1);
+  }
+
+  console.log("No differences found");
+}
+
+function connectToReadModelDb({
+  readModelDbHost: host,
+  readModelDbPort: port,
+  readModelDbUsername: username,
+  readModelDbPassword: password,
+  readModelDbName: database,
+}: ReadModelDbConfig): Db {
+  const mongoDBConnectionURI = `mongodb://${username}:${password}@${host}:${port}`;
+  const client = new MongoClient(mongoDBConnectionURI, {
+    retryWrites: false,
+  });
+  return client.db(database);
+}
+
+export async function getReadModelsCollectionDataDifferences(
+  readmodelA: Db,
+  readmodelB: Db,
+  collection: string
+): Promise<Array<[Identifiable, Identifiable]>> {
+  const resultsA = await readmodelA
+    .collection(collection)
+    .find()
+    .map(({ data }) => Identifiable.parse(data))
+    .toArray();
+
+  const resultsB = await readmodelB
+    .collection(collection)
+    .find({ id: { $in: resultsA.map((d) => d.id) } })
+    .map(({ data }) => Identifiable.parse(data))
+    .toArray();
+
+  return zipIdentifiableData(resultsA, resultsB).filter(
+    ([a, b]) => !isEqual(a, b)
+  );
```

Comment:
Is it important to check that both readmodels collections have the same amount of data? 
Here I'm not, I take all the data of the collection from readmodel A, and query the readmodelB filtering for the ids of from results of the first query.

Reply:
I think we should check that for every item in `readmodel Scala` there is an identical item in `readmodel TS` (by _identical_ I mean both id and properties are checked).
We should be able to find out errors like:
- `item1` exists in `readmodel Scala` but not in `readmodel TS`
- `item2` exists in `readmodel Scala` and in `readmodel TS`, but the properties are different
- `item3` exists in `readmodel TS` but not in `readmodel Scala`. This last case is important as well because we could identify if we didn't process a deletion event.

Reply:
Great, thanks for the hint! Working on it